src/libraries/AccountAccessParser.sol

```diff
@@ -239,6 +241,127 @@ library AccountAccessParser {
+                    else if (isGnosisSafeApproveHash(diff)) {
+                        shouldInclude = false;
+                    }
+                }
+                // 3. If the slot contains a timestamp, normalize it to zeroes.
+                else if (slotContainsTimestamp(account, diff)) {
+                    diff = normalizeTimestamp(diff);
+                }
+
+                // Include the diff in our normalized array if it should be included.
+                if (shouldInclude) {
+                    normalizedChanges[normalizedCount] = TempStateChange({
+                        who: account,
+                        slot: diff.slot,
+                        firstOld: diff.oldValue,
+                        lastNew: diff.newValue
+                    });
+                    normalizedCount++;
+                    require(normalizedCount < maxStateChanges, "AccountAccessParser: Max state changes reached");
+                }
+            }
+        }
+
+        // Create the final array with the correct size.
+        TempStateChange[] memory finalArray = new TempStateChange[](normalizedCount);
+        for (uint256 i = 0; i < normalizedCount; i++) {
+            finalArray[i] = normalizedChanges[i];
+        }
+
+        // Return keccak256 hash of the abi-encoded normalized array.
+        return keccak256(abi.encode(finalArray));
+    }
+
+    /// @notice Checks if the state diff represents an EOA nonce increment
+    function isEOANonceIncrement(address _account, StateDiff memory _diff) internal view returns (bool) {
+        uint256 codeSize = _account.code.length;
+        return codeSize == 0 && _diff.slot == bytes32(0) && uint256(_diff.newValue) == uint256(_diff.oldValue) + 1;
+    }
+
+    /// @notice Checks if the state diff represents a Gnosis Safe nonce increment
+    function isGnosisSafeNonceIncrement(StateDiff memory _diff) internal pure returns (bool) {
+        // In Gnosis Safe, the nonce is stored at slot 5. See `GnosisSafeStorage.sol` to verify.
+        return _diff.slot == GNOSIS_SAFE_NONCE_SLOT && uint256(_diff.newValue) == uint256(_diff.oldValue) + 1;
+    }
+
+    /// @notice Checks if the state diff represents setting an approve hash in a Gnosis Safe
+    function isGnosisSafeApproveHash(StateDiff memory _diff) internal pure returns (bool) {
+        // In Gnosis Safe, approvedHashes is a mapping at slot 8
+        // The slot for a specific hash approval is calculated with the keccak256 hash of the address
+        // and slot. We can't know the specific slot value here, but we can detect a change from 0 to 1
+        // which is a characteristic of an approve hash operation. We assume that if such a storage
+        // change is seen in a very large slot, it is an approve hash operation. Setting modules
+        // and guards in storage would not result in the slot having a value of 1.
+        return _diff.oldValue == bytes32(0) && _diff.newValue == bytes32(uint256(1))
```

Comment:
I'm thinking maybe you can compute the slot here by requiring the parent multisig address as an argument to the ``normalizedStateDiffHash` function. You can see I've implemented similar logic already in `printApproveHashInfo`.

Reply:
Can we leave this for a future PR or do you think this adds value? Looking at the safe storage layout, the only other mapping that sets slots to 1 is the `signedMessages` mapping but we don't use that. (The owners linked list can also have a value of 1 for the sentinel owner but that is only written when the safe is originally setup)

Reply:
Makes sense. Happy with either. If it's not implemented as part of this PR I think it need a 'TODO' then.

Reply:
Added in 51a674c0